lib/bolt/pal/yaml_plan/step/upload.rb

```diff
@@ -6,7 +6,7 @@ class YamlPlan
       class Step
         class Upload < Step
           def self.allowed_keys
-            super + Set['source', 'destination']
+            super + Set['source', 'destination', 'upload']
           end
 
           def self.required_keys
```

Comment:
No, https://github.com/puppetlabs/bolt/pull/1979/files#diff-ec96a76d58680201f59e7377e9d96cfaR96-R98 checks if the `upload` key is present and removes `source` as a required key if it is. This is similar to how we handle yaml plans that use the deprecated `target` key instead of `targets`.

I do think I got this a bit backwards by leaving `source` as the required key. Updated to have `upload` required instead.